remote.go

```diff
@@ -254,6 +254,8 @@ func credentialsCallback(_cred **C.git_cred, _url *C.char, _username_from_url *C
 		return C.GIT_EUSER
 	}
 	if cred != nil {
+		// we are transferring ownership to libgit, have to disable  
+		runtime.SetFinalizer(cred, nil)
```

Comment:
can this also do

```go
cred.ptr = nil
```

afterwards? this should prevent a (theoretical) use-after-free and will change it instead to an immediate crash which should aid in debuggability if the situation were to occur.

Reply:
OK, I can add that. But, `cred` is a local obj/var, it's not going to be used from any other places...

Reply:
theoretically the caller could still be holding a reference to the returned `cred` object from the callback. it's a bit paranoid, but it's still a possibility.